code/modules/reagents/reagent_containers/food/cans.dm

```diff
@@ -24,6 +29,159 @@
+							else
+								return
+					if("Remove")
+						if(!use_check_and_message(user))
+							user.visible_message("<b>[user]</b> removes the steel wool from \the [name].", "You remove the steel wool fuse from \the [name].")
+							FuseRemove()
+					if("Cancel")
+						return
+				return
+	
+	if(W.isFlameSource() && fuselength)
+		if(can_light())
+			fuselit = TRUE
+			if(reagents.get_reagent_amount(/datum/reagent/fuel) >= LETHAL_FUEL_CAPACITY && user)
+				msg_admin_attack("[user] ([user.ckey]) lit the fuse on an improvised [name] grenade. (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[user.x];Y=[user.y];Z=[user.z]'>JMP</a>)",ckey=key_name(user))
+			if(fuselength >= 3 && fuselength <= 5 && prob(30) && reagents.get_reagent_amount(/datum/reagent/fuel) >= LETHAL_FUEL_CAPACITY)
+				user.visible_message(SPAN_DANGER("<b>[user]</b> accidentally takes \the [W] too close to \the [name]'s opening!"))
+				detonate(TRUE) // it'd be a bit dull if the toy-levels of fuel had a chance to insta-pop, it's mostly just a way to keep the grenade balance in check
+			if(fuselength in list(1, 2))
+				user.visible_message(SPAN_DANGER("<b>[user]</b> tries to light the fuse on \the [name] but it was too short!"), SPAN_DANGER("You try to light the fuse but it was too short!"))
+				detonate(TRUE) // if you're somehow THAT determined and/or ignorant you managed to get the fuse below 3 seconds, so be it. reap what you sow.
+			else
+				user.visible_message(SPAN_WARNING("<b>[user]</b> lights the steel wool on \the [name] with \the [W]!"), SPAN_WARNING("You light the steel wool on \the [name] with the [W]!"))
+				playsound(get_turf(src), 'sound/items/flare.ogg', 50)
+				detonate(FALSE)
+	. = ..()
+
+/obj/item/reagent_containers/food/drinks/cans/proc/detonate(var/instant)
+	var/fuel = reagents.get_reagent_amount(/datum/reagent/fuel)
+	if(instant)
+		fuselength = 0
+	else if(prob(fuselength * 6)) // the longer the fuse, the higher chance it will fizzle out (18% chance minimum)
+		var/fizzle = rand(1, fuselength - 1)
+		sleep(fizzle * 10)
+		
+		fuselength -= fizzle
+		visible_message(SPAN_WARNING("The fuse on \the [name] fizzles out early."))
+		playsound(get_turf(src), 'sound/items/cigs_lighters/cig_snuff.ogg', 50)
+		fuselit = FALSE
+		update_icon()
+		return
+	else
+		fuselength += rand(-2, 2)
+	sleep(fuselength * 10)
+
+	switch(fuel)
+		if(0)
+			visible_message(SPAN_NOTICE("\The [name]'s fuse burns out and nothing happens."))
+			fuselit = FALSE
+			fuselength = 0
+			update_icon()
+		if(1 to 10) // baby explosion
+			var/obj/item/trash/can/popped_can = new /obj/item/trash/can(get_turf(src))
+			popped_can.icon_state = icon_state
+			popped_can.name = "popped can"
+			playsound(get_turf(src), 'sound/effects/snap.ogg', 50)
+			visible_message(SPAN_WARNING("\The [name] pops harmlessly!"))
+			fuselit = FALSE // shouldn't have to be done since it's going to get qdel'd, but in case something wacky happens in live play.
+			qdel(src)
```

Comment:
You have these two lines on three of the four switch statements. For less lines you could just have these two at the end and return from the first case.